lib/compress/zstdmt_compress.c

```diff
@@ -461,34 +461,36 @@ typedef struct {
     ZSTD_window_t ldmWindow;  /* A thread-safe copy of ldmState.window */
 } serialState_t;
 
-static int ZSTDMT_serialState_reset(serialState_t* serialState, ZSTDMT_seqPool* seqPool, ZSTD_CCtx_params params, size_t jobSize)
+static int ZSTDMT_serialState_reset(serialState_t* serialState, ZSTDMT_seqPool* seqPool, 
+                        ZSTD_CCtx_params* paramsPtr, size_t jobSize, const void* dict, 
```

Comment:
I presume that the change from `ZSTD_CCtx_params params` to `ZSTD_CCtx_params* paramsPtr`
reflects the fact that the content of `paramsPtr` can now be updated by this function,
and this update must be propagated beyond the lifetime of this function.

I also presume it's related to dictionary loading into ldmState.

It's a very important change, fundamentally changing a core property of this interface (namely that `params` was safe) so I guess it's worth underlining what needs to be updated, and why it's not possible to do it differently.